File: android/app/src/main/kotlin/com/goalmaker/app/application/auth/AuthSession.kt
package com.goalmaker.app.application.auth

/**
 * Whether someone is signed in. [Loading] lasts until the stored session has been read. [SignedIn]
 * carries the account's id, which owns every row the replica holds.
 */
sealed interface AuthSession {
    data object Loading : AuthSession

    data object SignedOut : AuthSession

    data class SignedIn(val userId: String, val email: String) : AuthSession
}

File: windows/src/GoalMaker.Core/Auth/AuthSession.cs
namespace GoalMaker.Core.Auth;

/// <summary>Whether someone is signed in. <see cref="Loading"/> lasts until the stored session is read.</summary>
public abstract record AuthSession
{
    private AuthSession()
    {
    }

    public sealed record Loading : AuthSession;

    public sealed record SignedOut : AuthSession;

    public sealed record SignedIn(string UserId, string Email) : AuthSession;
}
